### PR TITLE
False Positive From Version Check

### PR DESCRIPTION
**Pull request by OwenArnold**
Ultimately we are going to move our hdf5 library forward, and that is the subject of a separate dependency management thread that @martyngigg is running.

hdf5 operates like multiple version in one - Generally there is poor compatibility between 1.8 and 1.10. Files written in 1.10 are not guaranteed to work with 1.8. We have tried to identify which features may not be supported and stop execution if we detect incompatibility between library versions - however there are cases where the check can give a false positive, namely if `hdf5format_compat` has been run on the input files.

The change here demotes the exception to a warning. In order for the logging (deliberate side-effect) to work sensibly. I have reorganised some of the loggable free-functions into an internal class, otherwise there would be excessive parameterisation of free functions. Note that I have deliberately tried to decouple the logging from Mantid's internal logging/poco functionality.

### CODE REVIEW

File: Framework/NexusGeometry/inc/MantidNexusGeometry/NexusGeometryParser.h [hunk 1]
```diff
@@ -16,12 +16,44 @@ namespace Geometry {
...
 namespace NexusGeometry {
-/** NexusGeometryParser : Responsible for parsing a nexus geometry file and
+
+/**
+ * Abstract logger. Avoid hard-coded logging dependencies.
+ */
+class MANTID_NEXUSGEOMETRY_DLL Logger {
+public:
+  virtual void warning(const std::string &warning) = 0;
+  virtual void error(const std::string &warning) = 0;
```
Comment (nvaytet) [inline Framework/NexusGeometry/inc/MantidNexusGeometry/NexusGeometryParser.h]:
```suggestion
  virtual void error(const std::string &error) = 0;
```

**Review by nvaytet — CHANGES_REQUESTED** [review]
Small thing: Strange: why was the `images/mantid_notebook_unstable.ico` file renamed?

Apart from that, all looks good.

**Comment by OwenArnold** [comment]
> Small thing: Strange: why was the `images/mantid_notebook_unstable.ico` file renamed?
> 
> Apart from that, all looks good.

That was a problem in Master at the time. I'll remerge master.

**Review by nvaytet — APPROVED** [review]

**Comment by nvaytet** [comment]
Hi Owen, a new warning has popped up with `clang` after you merged the `disable_version_check2` branch. Can you check what happened?

**Pull request merged**